File: core/cpp/src/main/vx/translation/es.cpp
#include "../../vx/core.hpp"
#include "../../vx/translate.hpp"
#include "es.hpp"

namespace vx_translation_es {



  // (func translation-es)
  vx_core::Type_translation f_translation_es() {
    vx_core::Type_translation output = vx_core::e_translation;
    output = vx_core::f_new(
      vx_core::t_translation,
      vx_core::vx_new(vx_core::t_anylist, {
        vx_core::vx_new_string(":name"),
        vx_core::vx_new_string("es"),
        vx_core::vx_new_string(":wordmap"),
        vx_translation_es::f_words()
      })
    );
    return output;
  }

  // (func translation-es)
  // class Class_translation_es {
    Abstract_translation_es::~Abstract_translation_es() {}

    Class_translation_es::Class_translation_es() : Abstract_translation_es::Abstract_translation_es() {
      vx_core::refcount += 1;
    }

    Class_translation_es::~Class_translation_es() {
      vx_core::refcount -= 1;
      if (this->vx_p_msgblock) {
        vx_core::vx_release_one(this->vx_p_msgblock);
      }
    }

    vx_core::Type_any Class_translation_es::vx_new(vx_core::vx_Type_listany vals) const {
      vx_translation_es::Func_translation_es output = vx_translation_es::e_translation_es;
      vx_core::vx_release(vals);
      return output;
    }

    vx_core::Type_any Class_translation_es::vx_copy(vx_core::Type_any copyval, vx_core::vx_Type_listany vals) const {
      vx_translation_es::Func_translation_es output = vx_translation_es::e_translation_es;
      vx_core::vx_release_except(copyval, output);
      vx_core::vx_release_except(vals, output);
      return output;
    }

    vx_core::Type_typedef Class_translation_es::vx_typedef() const {
      vx_core::Type_typedef output = vx_core::Class_typedef::vx_typedef_new(
        "vx/translation/es", // pkgname
        "translation-es", // name
        ":func", // extends
        vx_core::vx_new(vx_core::t_typelist, {vx_core::t_func}), // traits
        vx_core::e_typelist, // allowtypes
        vx_core::e_typelist, // disallowtypes
        vx_core::e_funclist, // allowfuncs
        vx_core::e_funclist, // disallowfuncs
        vx_core::e_anylist, // allowvalues
        vx_core::e_anylist, // disallowvalues
        vx_core::e_argmap // properties
      );
      return output;
    }

    vx_core::Type_constdef Class_translation_es::vx_constdef() const {return this->vx_p_constdef;}

    vx_core::Type_funcdef Class_translation_es::vx_funcdef() const {
      vx_core::Type_funcdef output = vx_core::Class_funcdef::vx_funcdef_new(
        "vx/translation/es", // pkgname
        "translation-es", // name
        0, // idx
        false, // async
        this->vx_typedef() // typedef
      );
      return output;
    }

    vx_core::Type_any Class_translation_es::vx_empty() const {return vx_translation_es::e_translation_es;}
    vx_core::Type_any Class_translation_es::vx_type() const {return vx_translation_es::t_translation_es;}
    vx_core::Type_msgblock Class_translation_es::vx_msgblock() const {return this->vx_p_msgblock;}
    vx_core::vx_Type_listany Class_translation_es::vx_dispose() {return vx_core::emptylistany;}

    vx_core::Type_any Class_translation_es::vx_repl(vx_core::Type_anylist arglist) {
      vx_core::Type_any output = vx_core::e_any;
      output = vx_translation_es::f_translation_es();
      vx_core::vx_release_except(arglist, output);
      return output;
    }

  //}

  // (func words)
  vx_core::Type_stringmap f_words() {
    vx_core::Type_stringmap output = vx_core::e_stringmap;
    output = vx_core::f_new(
      vx_core::t_stringmap,
      vx_core::vx_new(vx_core::t_anylist, {
        vx_core::vx_new_string("Hello"),
        vx_core::vx_new_string("Hola"),
        vx_core::vx_new_string("World"),
        vx_core::vx_new_string("Mundo")
      })
    );
    return output;
  }

  // (func words)
  // class Class_words {
    Abstract_words::~Abstract_words() {}

    Class_words::Class_words() : Abstract_words::Abstract_words() {
      vx_core::refcount += 1;
    }

    Class_words::~Class_words() {
      vx_core::refcount -= 1;
      if (this->vx_p_msgblock) {
        vx_core::vx_release_one(this->vx_p_msgblock);
      }
    }

    vx_core::Type_any Class_words::vx_new(vx_core::vx_Type_listany vals) const {
      vx_translation_es::Func_words output = vx_translation_es::e_words;
      vx_core::vx_release(vals);
      return output;
    }

    vx_core::Type_any Class_words::vx_copy(vx_core::Type_any copyval, vx_core::vx_Type_listany vals) const {
      vx_translation_es::Func_words output = vx_translation_es::e_words;
      vx_core::vx_release_except(copyval, output);
      vx_core::vx_release_except(vals, output);
      return output;
    }

    vx_core::Type_typedef Class_words::vx_typedef() const {
      vx_core::Type_typedef output = vx_core::Class_typedef::vx_typedef_new(
        "vx/translation/es", // pkgname
        "words", // name
        ":func", // extends
        vx_core::vx_new(vx_core::t_typelist, {vx_core::t_func}), // traits
        vx_core::e_typelist, // allowtypes
        vx_core::e_typelist, // disallowtypes
        vx_core::e_funclist, // allowfuncs
        vx_core::e_funclist, // disallowfuncs
        vx_core::e_anylist, // allowvalues
        vx_core::e_anylist, // disallowvalues
        vx_core::e_argmap // properties
      );
      return output;
    }

    vx_core::Type_constdef Class_words::vx_constdef() const {return this->vx_p_constdef;}

    vx_core::Type_funcdef Class_words::vx_funcdef() const {
      vx_core::Type_funcdef output = vx_core::Class_funcdef::vx_funcdef_new(
        "vx/translation/es", // pkgname
        "words", // name
        0, // idx
        false, // async
        this->vx_typedef() // typedef
      );
      return output;
    }

    vx_core::Type_any Class_words::vx_empty() const {return vx_translation_es::e_words;}
    vx_core::Type_any Class_words::vx_type() const {return vx_translation_es::t_words;}
    vx_core::Type_msgblock Class_words::vx_msgblock() const {return this->vx_p_msgblock;}
    vx_core::vx_Type_listany Class_words::vx_dispose() {return vx_core::emptylistany;}

    vx_core::Type_any Class_words::vx_repl(vx_core::Type_anylist arglist) {
      vx_core::Type_any output = vx_core::e_any;
      output = vx_translation_es::f_words();
      vx_core::vx_release_except(arglist, output);
      return output;
    }

  //}

  vx_translation_es::Func_translation_es e_translation_es = NULL;
  vx_translation_es::Func_translation_es t_translation_es = NULL;
  vx_translation_es::Func_words e_words = NULL;
  vx_translation_es::Func_words t_words = NULL;

  // class vx_Class_package {
    vx_Class_package::vx_Class_package() {
      init();
    }
    void vx_Class_package::init() {
      vx_translation_es::e_translation_es = new vx_translation_es::Class_translation_es();
      vx_core::vx_reserve_empty(vx_translation_es::e_translation_es);
      vx_translation_es::t_translation_es = new vx_translation_es::Class_translation_es();
      vx_core::vx_reserve_type(vx_translation_es::t_translation_es);
      vx_translation_es::e_words = new vx_translation_es::Class_words();
      vx_core::vx_reserve_empty(vx_translation_es::e_words);
      vx_translation_es::t_words = new vx_translation_es::Class_words();
      vx_core::vx_reserve_type(vx_translation_es::t_words);
      vx_core::vx_Type_mapany maptype;
      vx_core::vx_Type_mapany mapconst;
      vx_core::vx_Type_mapfunc mapfunc;
      vx_core::vx_Type_mapany mapempty;
      mapfunc["translation-es"] = vx_translation_es::t_translation_es;
      mapfunc["words"] = vx_translation_es::t_words;
      vx_core::vx_global_package_set("vx/translation/es", maptype, mapconst, mapfunc);
	   }
  // }

}

File: core/csharp/src/vx/translation/Es.cs
namespace Vx.Translation;

public static class Es {

  /**
   * @function translation_es
   * Returns translation data
   * @return {translation}
   * (func translation-es)
   */
  public interface Func_translation_es : Vx.Core.Type_func, Vx.Core.Type_replfunc {
    public Vx.Core.Type_translation vx_translation_es();
  }

  public class Class_translation_es : Vx.Core.Class_base, Func_translation_es {

    public override Vx.Translation.Es.Func_translation_es vx_new(params Object[] vals) {
      Class_translation_es output = new Class_translation_es();
      return output;
    }

    public override Vx.Translation.Es.Func_translation_es vx_copy(params Object[] vals) {
      Class_translation_es output = new Class_translation_es();
      return output;
    }

    public override Vx.Core.Type_typedef vx_typedef() {
      return Vx.Core.t_func.vx_typedef();
    }

    public Vx.Core.Type_funcdef vx_funcdef() {
      return Vx.Core.funcdef_new(
        "vx/translation/es", // pkgname
        "translation-es", // name
        0, // idx
        false, // async
        Vx.Core.typedef_new(
          "vx/core", // pkgname
          "translation", // name
          ":struct", // extends
          Vx.Core.e_typelist, // traits
          Vx.Core.e_typelist, // allowtypes
          Vx.Core.e_typelist, // disallowtypes
          Vx.Core.e_funclist, // allowfuncs
          Vx.Core.e_funclist, // disallowfuncs
          Vx.Core.e_anylist, // allowvalues
          Vx.Core.e_anylist, // disallowvalues
          Vx.Core.e_argmap // properties
        ) // typedef
      );
    }

    public override Vx.Core.Type_any vx_empty() {
      return e_translation_es;
    }

    public override Vx.Core.Type_any vx_type() {
      return t_translation_es;
    }

    public Vx.Core.Type_any vx_repl(Vx.Core.Type_anylist arglist) {
      Vx.Core.Type_any output = Vx.Core.e_any;
      output = Vx.Translation.Es.f_translation_es();
      return output;
    }

    public Vx.Core.Type_translation vx_translation_es() {
      return Vx.Translation.Es.f_translation_es();
    }

  }

  public static Func_translation_es e_translation_es = new Vx.Translation.Es.Class_translation_es();
  public static Func_translation_es t_translation_es = new Vx.Translation.Es.Class_translation_es();

  public static Vx.Core.Type_translation f_translation_es() {
    Vx.Core.Type_translation output = Vx.Core.e_translation;
    output = Vx.Core.f_new(
      Vx.Core.t_translation,
      Vx.Core.vx_new(Vx.Core.t_anylist,
        Vx.Core.vx_new_string(":name"),
        Vx.Core.vx_new_string("es"),
        Vx.Core.vx_new_string(":wordmap"),
        Vx.Translation.Es.f_words()
      )
    );
    return output;
  }

  /**
   * @function words
   * Word Translation
   * @return {stringmap}
   * (func words)
   */
  public interface Func_words : Vx.Core.Type_func, Vx.Core.Type_replfunc {
    public Vx.Core.Type_stringmap vx_words();
  }

  public class Class_words : Vx.Core.Class_base, Func_words {

    public override Vx.Translation.Es.Func_words vx_new(params Object[] vals) {
      Class_words output = new Class_words();
      return output;
    }

    public override Vx.Translation.Es.Func_words vx_copy(params Object[] vals) {
      Class_words output = new Class_words();
      return output;
    }

    public override Vx.Core.Type_typedef vx_typedef() {
      return Vx.Core.t_func.vx_typedef();
    }

    public Vx.Core.Type_funcdef vx_funcdef() {
      return Vx.Core.funcdef_new(
        "vx/translation/es", // pkgname
        "words", // name
        0, // idx
        false, // async
        Vx.Core.typedef_new(
          "vx/core", // pkgname
          "stringmap", // name
          ":map", // extends
          Vx.Core.e_typelist, // traits
          Vx.Core.vx_new(Vx.Core.t_typelist, Vx.Core.t_string), // allowtypes
          Vx.Core.e_typelist, // disallowtypes
          Vx.Core.e_funclist, // allowfuncs
          Vx.Core.e_funclist, // disallowfuncs
          Vx.Core.e_anylist, // allowvalues
          Vx.Core.e_anylist, // disallowvalues
          Vx.Core.e_argmap // properties
        ) // typedef
      );
    }

    public override Vx.Core.Type_any vx_empty() {
      return e_words;
    }

    public override Vx.Core.Type_any vx_type() {
      return t_words;
    }

    public Vx.Core.Type_any vx_repl(Vx.Core.Type_anylist arglist) {
      Vx.Core.Type_any output = Vx.Core.e_any;
      output = Vx.Translation.Es.f_words();
      return output;
    }

    public Vx.Core.Type_stringmap vx_words() {
      return Vx.Translation.Es.f_words();
    }

  }

  public static Func_words e_words = new Vx.Translation.Es.Class_words();
  public static Func_words t_words = new Vx.Translation.Es.Class_words();

  public static Vx.Core.Type_stringmap f_words() {
    Vx.Core.Type_stringmap output = Vx.Core.e_stringmap;
    output = Vx.Core.f_new(
      Vx.Core.t_stringmap,
      Vx.Core.vx_new(Vx.Core.t_anylist,
        Vx.Core.vx_new_string("Hello"),
        Vx.Core.vx_new_string("Hola"),
        Vx.Core.vx_new_string("World"),
        Vx.Core.vx_new_string("Mundo")
      )
    );
    return output;
  }


  public static class PackageRunOnce {
    public static void RunOnce() {
    Vx.Core.Map<string, Vx.Core.Type_any> maptype = new Vx.Core.LinkedHashMap<string, Vx.Core.Type_any>();
    Vx.Core.Map<string, Vx.Core.Type_any> mapconst = new Vx.Core.LinkedHashMap<string, Vx.Core.Type_any>();
    Vx.Core.Map<string, Vx.Core.Type_func> mapfunc = new Vx.Core.LinkedHashMap<string, Vx.Core.Type_func>();
    mapfunc.put("translation-es", Vx.Translation.Es.t_translation_es);
    mapfunc.put("words", Vx.Translation.Es.t_words);
    Vx.Core.vx_global_package_set("vx/translation/es", maptype, mapconst, mapfunc);
    }
  }

}
